core/tests/protractor_desktop/coreEditorAndPlayerFeatures.js

```diff
@@ -309,10 +312,50 @@ describe('Core exploration functionality', function() {
       richTextEditor.appendUnorderedList(['an entry', 'another entry']);
     });
     explorationEditorMainTab.setInteraction(
-      'MultipleChoiceInput',
-      [forms.toRichText('option A'), forms.toRichText('option B')]);
+      'MultipleChoiceInput', [{ editAction: 'create'},
+        [forms.toRichText('option A'), forms.toRichText('option B'),
+          forms.toRichText('option C'), forms.toRichText('option D'),
+          forms.toRichText('option E'), forms.toRichText('option F')]]);
+    explorationEditorMainTab.addResponse('MultipleChoiceInput',
+      function(richTextEditor) {
+        richTextEditor.appendBoldText('correct');
+      }, 'final card', true, 'Equals', 'option A');
+    explorationEditorMainTab.getResponseEditor(0)
+      .addRule('MultipleChoiceInput', 'Equals', 'option C');
+    explorationEditorMainTab.getResponseEditor(0)
+      .addRule('MultipleChoiceInput', 'Equals', 'option F');
+    explorationEditorMainTab.getResponseEditor(0)
+      .expectFeedbackInstructionToBe('correct');
+
+    explorationEditorMainTab.getResponseEditor(0).expectMultipleRulesToBe(
+      [['MultipleChoiceInput', 'Equals', ['option A']],
+        ['MultipleChoiceInput', 'Equals', ['option C']],
+        ['MultipleChoiceInput', 'Equals', ['option F']]]);
+    // delete the options corresponding to answer rules.
+    element(by.css('.protractor-test-interaction')).click();
+    explorationEditorMainTab.editInteraction(
+      'MultipleChoiceInput', [{editAction: 'delete'}, [0, 1, 3]]);
+    // add 3 more options to the interaction.
+    element(by.css('.protractor-test-interaction')).click();
+    explorationEditorMainTab.editInteraction(
+      'MultipleChoiceInput', [{editAction: 'add'},
+        [forms.toRichText('option G '),
+          forms.toRichText('option H'), forms.toRichText('option I')]]);
+    // validates that the answer rules are invalid if the corresponding option
+    // is deleted
```

Comment:
can you also explain that you added the 3 more options above to test that the answer rules would be invalid even if the answers were replaced (making the indices valid again)?